File: Patterns/Strategy/java/src/heroes/Hero.java
package heroes;

import weapons.interfaces.IWeaponBehavior;

public abstract class Hero {
	
	String name = "NoName";
	IWeaponBehavior currentWeapon;
	
	public Hero(){
	}
	
	public void setWeapon(IWeaponBehavior weapon){
		currentWeapon = weapon;
	}
	
	public void attack(){
		System.out.print(name + " attacked with: ");
		currentWeapon.performAttack();
	}
	
}

File: Patterns/Strategy/php/www/heroes/Hero.php
<?php
	require_once 'weapons/interfaces/IWeaponBehavior.php';
	class Hero{
		
		public $currentWeapon;
		public $name;
		
		public function attack(){
			
			echo $this->name . " attacked with: ";
			$this->currentWeapon->performAttack();			
		}
		
		public function setWeapon(IWeaponBehavior $weapon){
			$this->currentWeapon = $weapon;
		}
		
		
	}